src/main/kotlin/com/openlattice/indexing/BackgroundLinkingIndexingService.kt

```diff
@@ -114,13 +110,13 @@ class BackgroundLinkingIndexingService(
     @Suppress("UNUSED")
     @Scheduled(fixedRate = LINKING_INDEX_RATE)
     fun updateCandidateList() {
-        if ( !indexerConfiguration.backgroundLinkingIndexingEnabled ) {
+        if (!indexerConfiguration.backgroundLinkingIndexingEnabled) {
             return
         }
         executor.submit {
             logger.info("Registering linking ids needing indexing.")
 
-            candidates.addAll( getDirtyLinkingIds() )
+            candidates.addAll(getDirtyLinkingIds())
```

Comment:
I'm not sure this is safe given that the LIMIT was removed. Since this will likely be adding essentially all the person entities in the system to this queue, it seems like this could explode memory or hold postgres connections open for a really long time causing us to run out, and could also possibly re-queue the same entities multiple times if linker can't get through the entire queue in the `LINKING_INDEX_RATE ` time until another queueing job spawns

Reply:
Yeah, true! Fetch size will only limit loading size, but addAll probably loads all of them into the queue. Will change it back to limit the query results

Reply:
Given the design of indexer I don't think holding postgres connections open for too long should be an issue (as long as we know the max total number of connections that might be open (which should be fairly low). 

Given that we're using a queue and producer/consumer model here, we can assume that every indexing process holds a connection open where the driver is doing batches of fetch size element reads and attempting to insert them into a blocking queue (make sure you use ```put(...)```) which will pause reading from the database (hence avoiding out of memory issues). On the read size we will use a connection to read the actual data and mark things as indexed. It's fine if there is some contention here as there will be natural back-pressure to prevent more queuing. 

Two more considerations-- you can control the max elements in the queue and you can also index in batch (by storing batches of id in the queue instead of a single id at a time). If you want to avoid implementing another stream serializer you can poll the queue until you have your batch size or no elements are available, then index.